### ISSUE TITLE
Can't bind to 'schema' since it isn't a known property of 'sf-form'

### ISSUE THREAD

**Issue by suvarna-tammina** [makinacorpus/angular2-schema-form#59]
Can you please help me why i am getting issue. By seeing your previous closed issue i have tried to  replace   schema-form by sf-form in my template but still i am facing problem.
![error](https://cloud.githubusercontent.com/assets/21304003/22504647/631d675e-e89d-11e6-8776-72d4730c1110.png)

**Comment by rgzr** [comment]
Try to import with `/src` appended to your import path.

`import { SchemaFormModule, WidgetRegistry, DefaultWidgetRegistry } from "angular2-schema-form/src";`

instead of:

`import { SchemaFormModule, WidgetRegistry, DefaultWidgetRegistry } from "angular2-schema-form";`

**Comment by fbessou** [comment]
The above configuration should work and does not need AOT compilation. Have you added the SchemaFormModule to the `imports` section of your `app.module.ts` (or equivalent) ?

```typescript
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';

import {
  DefaultWidgetRegistry,
  SchemaFormModule,
  WidgetRegistry
} from 'angular2-schema-form/src';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    SchemaFormModule
  ],
  providers: [{provide: WidgetRegistry, useClass: DefaultWidgetRegistry}],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

**Comment by suvarna-tammina** [comment]
ya i have added all those in app.module.ts.
In my component i have added example json schema then after in  html template when i am trying to add sf-form tag like this
<sf-form [schema]="mySchema" ></sf-form>. I am getting issue.

**Comment by fbessou** [comment]
Which version of angular2-schema-form are you using ?

**Comment by suvarna-tammina** [comment]
'angular2-schema-form':1.0.0-alpha.33

**Comment by fbessou** [comment]
You don't get any error in your terminal when you build your project ?

**Comment by suvarna-tammina** [comment]
No. I am not getting any errors in the terminal while building the project. The only error i am getting is when try to keep the <sf-form> </sf-form> tag  in the template i am getting the error. Don't know Somewhere i am missing but this is creating a problem for me.

**Comment by rgzr** [comment]
This happened to me when not using `/src` import path since i use webpack. 

Is `app.module.ts` the only module on your project? Maybe you have to do the imports in the module where you are using the form.

**Comment by suvarna-tammina** [comment]
I have different modules in my project not only app.module.ts. As you suggested  i have imported SchemaFormModule with /src appended to the path to the imports in app.module.ts  then the cannot find module  error gone.
secondly, I even added the SchemaFormModule  in the imports of sub module in my project also.  Please suggest me how to make <sf-form> tag to work?

**Comment by fbessou** [comment]
Does the demo work for you ? https://github.com/fbessou/angular2-schema-form-demo

**Comment by sergeik** [comment]
Same error.
I use https://github.com/AngularClass/angular2-webpack-starter/ as base for project.

Demo https://github.com/fbessou/angular2-schema-form-demo works well

**Comment by suvarna-tammina** [comment]
demo was working perfectly.......

**Comment by TheDarkX** [comment]
Same as @suvarna-tammina 
Demo was working fine. But implement into project gets such error.

![image](https://cloud.githubusercontent.com/assets/7087449/22584236/449cbab0-ea2c-11e6-95a3-18ffa23d6381.png)

I tried npm cache clean and rebuild the project, but no luck.
It works fine with <schema-form> tag with no /src
When implement into project with <sf-form> tag + /src it shows above attached error.

**Comment by fbessou** [comment]
@suvarna-tammina: What is the project base you are using?

**Comment by suvarna-tammina** [comment]
Our project is created using angular-cli and it is webpack based. Is that information your were asking?

**Comment by TheDarkX** [comment]
Using <sf-form> tag and import path with /src
The demo working fine. But when trying to implement in my project then hits error

![image](https://cloud.githubusercontent.com/assets/7087449/22634003/dcd40d84-ec60-11e6-8ba4-c28623331dac.png)

**Comment by TheDarkX** [comment]
The demo working fine. But when implement into my project it hits error

Using <sf-form> tag and import path with /src
![untitled](https://cloud.githubusercontent.com/assets/7087449/22634056/53546724-ec61-11e6-8109-1d58a0ab12e3.png)

**Comment by cyrilzaharchenko** [comment]
I had the same issue yesterday. I managed to get it working by importing the module from 'src' like this:

```import { SchemaFormModule, WidgetRegistry, DefaultWidgetRegistry } from 'angular2-schema-form/src';```

Another thing is that it should be imported to the module where it will be used, not just in the root module.
And I've noticed several ```console.log``` messages coming from this project. Would be great if they will be removed.

**Comment by Ragularuban** [comment]
Guys, in the html instead of 
`<sf-form [schema]="mySchema" [model]="myModel"></sf-form>`
Use 
`<schema-form [schema]="mySchema" [model]="myModel"></schema-form>`

It solves the problem

**Comment by i-am-dror** [comment]
getting the same problem for both:
<sf-form [schema]="mySchema" [model]="myModel"></sf-form>
<schema-form [schema]="mySchema" [model]="myModel"></schema-form>

and angular2-schema-form folder has only dist, no src folder.

**Comment by bojue** [comment]
ERROR Error: Uncaught (in promise): Error: Template parse errors:
Can't bind to 'schema' since it isn't a known property of 'sf-form'.
1. If 'sf-form' is an Angular component and it has 'schema' input, then verify that it is part of this module.
2. If 'sf-form' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("<sf-form [ERROR ->][schema]="mySchema"></sf-form>"): ng:///AdminModule/AccountSchemaComponent.html@0:9
'sf-form' is not a known element:
1. If 'sf-form' is an Angular component, then verify that it is part of this module.
2. If 'sf-form' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("[ERROR ->]<sf-form [schema]="mySchema"></sf-form>"): ng:///AdminModule/AccountSchemaComponent.html@0:0
Error: Template parse errors:
Can't bind to 'schema' since it isn't a known property of 'sf-form'.
1. If 'sf-form' is an Angular component and it has 'schema' input, then verify that it is part of this module.
2. If 'sf-form' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("<sf-form [ERROR ->][schema]="mySchema"></sf-form>"): ng:///AdminModule/AccountSchemaComponent.html@0:9
'sf-form' is not a known element:
1. If 'sf-form' is an Angular component, then verify that it is part of this module.
2. If 'sf-form' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("[ERROR ->]<sf-form [schema]="mySchema"></sf-form>"): ng:///AdminModule/AccountSchemaComponent.html@0:0

**Comment by Iquebal** [comment]
'sf-form' is not a known element getting issues in angular 7
![issues](https://user-images.githubusercontent.com/8733452/54356182-00396e00-4681-11e9-8aab-2b93aaf8cf03.PNG)

**Comment by Iquebal** [comment]
> Guys, in the html instead of
> `<sf-form [schema]="mySchema" [model]="myModel"></sf-form>`
> Use
> `<schema-form [schema]="mySchema" [model]="myModel"></schema-form>`
> 
> It solves the problem

Not working
